src/deps.ts

```diff
@@ -296,13 +296,36 @@ export interface AutoEncryptionOptions {
     /** Command line arguments to use when auto-spawning a mongocryptd */
     mongocryptdSpawnArgs?: string[];
     /**
-     * Full path to a CSFLE shared library to be used (instead of mongocryptd)
-     * @experimental
+     * Full path to a CSFLE shared library to be used (instead of mongocryptd).
+     *
+     * This needs to be the path to the file itself, not a directory.
+     * It can be an absolute or relative path. If the path is relative and
+     * its first component is `$ORIGIN`, it will be replaced by the directory
+     * containing the mongodb-client-encryption native addon file. Otherwise,
+     * the path will be interpreted relative to the current working directory.
+     *
+     * Currently, loading different CSFLE shared library files from different
+     * MongoClients in the same process is not supported.
+     *
+     * If this option is provided and no CSFLE shared library could be loaded
+     * from the specified location, creating the MongoClient will fail.
```

Comment:
I believe connecting a MongoClient will fail since that's when we call the `autoEncrypter` init functions. I think it's worth making the distinction in case folks want to check for an error

Reply:
Creating the libmongocrypt handle is part of creating `AutoEncrypter`, which in turn is parts of options parsing. This is the stack trace:

```
TypeError: A CSFLE override path was specified [/tmp/blah.so], but we failed to open a dynamic library at that location
    at new AutoEncrypter (/home/addaleax/src/libmongocrypt/bindings/node/lib/autoEncrypter.js:135:26)
    at new Encrypter (/home/addaleax/src/libmongocrypt/bindings/node/node_modules/mongodb/lib/encrypter.js:48:30)
    at parseOptions (/home/addaleax/src/libmongocrypt/bindings/node/node_modules/mongodb/lib/connection_string.js:330:34)
    at new MongoClient (/home/addaleax/src/libmongocrypt/bindings/node/node_modules/mongodb/lib/mongo_client.js:62:63)
```

Reply:
Oh perfect, good to know, I think the earlier the failure the better anyway 🚀